packages/swatch/src/Swatch.ts

```diff
@@ -0,0 +1,243 @@
+    PropertyValues,
+    SizedMixin,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import { property } from '@spectrum-web-components/base/src/decorators.js';
+import { when } from '@spectrum-web-components/base/src/directives.js';
+import { Focusable } from '@spectrum-web-components/shared/src/focusable.js';
+import '@spectrum-web-components/icons-ui/icons/sp-icon-dash75.js';
+import '@spectrum-web-components/icons-ui/icons/sp-icon-dash100.js';
+import '@spectrum-web-components/icons-ui/icons/sp-icon-dash200.js';
+import '@spectrum-web-components/icons-ui/icons/sp-icon-dash300.js';
+
+import styles from './swatch.css.js';
+import dashStyles from '@spectrum-web-components/icon/src/spectrum-icon-dash.css.js';
+
+export type SwatchBorder = 'light' | 'none' | undefined;
+export type SwatchRounding = 'none' | 'full' | undefined;
+export type SwatchShape = 'rectangle' | undefined;
+
+const dashIcon: Record<string, () => TemplateResult> = {
+    xs: () => html`
+        <sp-icon-dash75
+            slot="icon"
+            class="mixedValueIcon spectrum-UIIcon-Dash75"
+        ></sp-icon-dash75>
+    `,
+    s: () => html`
+        <sp-icon-dash100
+            slot="icon"
+            class="mixedValueIcon spectrum-UIIcon-Dash100"
+        ></sp-icon-dash100>
+    `,
+    m: () => html`
+        <sp-icon-dash200
+            slot="icon"
+            class="mixedValueIcon spectrum-UIIcon-Dash200"
+        ></sp-icon-dash200>
+    `,
+    l: () => html`
+        <sp-icon-dash300
+            slot="icon"
+            class="mixedValueIcon spectrum-UIIcon-Dash300"
+        ></sp-icon-dash300>
+    `,
+};
+
+/**
+ * @element sp-swatch
+ */
+export class Swatch extends SizedMixin(Focusable, {
+    validSizes: ['xs', 's', 'm', 'l'],
+}) {
+    public static override get styles(): CSSResultArray {
+        return [styles, dashStyles];
+    }
+
+    @property({ reflect: true })
+    public border: SwatchBorder;
+
+    @property()
+    public color = '';
+
+    @property()
+    public label = '';
+
+    @property({ type: Boolean, reflect: true, attribute: 'mixed-value' })
+    public mixedValue = false;
+
+    @property({ type: Boolean, reflect: true })
+    public nothing = false;
+
+    @property({ reflect: true })
+    public role = 'button';
+
+    @property({ reflect: true })
+    public rounding: SwatchRounding;
+
+    @property({ type: Boolean, reflect: true })
+    public selected = false;
+
+    @property({ reflect: true })
+    public shape: SwatchShape;
+
+    @property()
+    get value(): string {
+        return this._value || this.color || this.label;
+    }
+
+    set value(value: string) {
+        if (value === this._value) return;
+        const oldValue = this.value;
+        this._value = value;
+        this.requestUpdate('value', oldValue);
+    }
+
+    private _value?: string;
+
+    public override get focusElement(): HTMLElement {
+        return this;
+    }
+
+    public toggle(destination?: boolean): void {
```

Comment:
Better named a `force`, like https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute#force? Rather than toggle one way or another, if present, it toggles to that value.

Reply:
I thought it might be intended as `force`, but it will only force to `true`, not `false`, if the swatch isn't selected. If we want to force `this.selected` to be the value of `force`, then toggle should be:

```ts
public toggle(force?: boolean): void {
  this.selected = force ?? !this.selected;
}
```